sympy/solvers/tests/test_inequalities.py

```diff
@@ -323,16 +340,24 @@ def test_issue_10198():
         -1 + 1/abs(1/x - 1) < 0) == Or(
         And(-oo < x, x < 0), And(S(0) < x, x < S(1)/2)
         )
-    raises(NotImplementedError, lambda: reduce_inequalities(abs(1/sqrt(x)) - 1, x))
+    assert reduce_inequalities(abs(1/sqrt(x)) - 1, x) == Eq(x, 1)
     assert reduce_abs_inequality(-3 + 1/abs(1 - 1/x), '<', x) == \
         Or(And(-oo < x, x < 0),
         And(S(0) < x, x < S(3)/4), And(S(3)/2 < x, x < oo))
     raises(ValueError,lambda: reduce_abs_inequality(-3 + 1/abs(
         1 - 1/sqrt(x)), '<', x))
 
+
 def test_issue_10047():
-    assert solve(sin(x) < 2) == And(-oo < x, x < oo)
+    assert solve(sin(x) < 2) == And(S(0) <= x, x < 2*pi)
```

Comment:
> To represent this particular answer we don't need Infinite union.

For solving trigonometric inequalities, we can return an infinite union.
In this particular case, the above-said union simplifies to `S.Reals`.
 
> Why do we need to return a wrong result in the first place?

I agree that this looks to be a regression. 
But in the long run, I don't think XFAILing this (for now) will cause us much harm.